asteroid/engine/gan_system.py

```diff
@@ -0,0 +1,161 @@
+from collections import OrderedDict
+import soundfile as sf
+import pytorch_lightning as pl
+import torch
+import os
+
+
+class GanSystem(pl.LightningModule):
+
+    def __init__(self, discriminator, generator, opt_d, opt_g,
+                 scheduler_d, scheduler_g, discriminator_loss, generator_loss,
+                 validation_loss, train_loader, val_loader=None, conf=None):
+
+        super(GanSystem, self).__init__()
+        self.discriminator = discriminator
+        self.generator = generator
+        self.opt_d = opt_d
+        self.opt_g = opt_g
+        self.d_loss = discriminator_loss
+        self.g_loss = generator_loss
+        self.validation_loss = validation_loss
+        self.train_loader = train_loader
+        self.val_loader = val_loader
+        self.scheduler_d = scheduler_d
+        self.scheduler_g = scheduler_g
+        self.conf = conf
+
+    def forward(self, z):
+        return self.generator(z)
+
+    def routine_step(self, batch, optimizer_idx):
+
+        # Get data from data_loader
+        inputs, targets = batch
+        # Forward inputs
+        estimates = self(inputs)
+        # The discriminator is basically a binary classifier
+        true_labels = True
+        false_labels = False
```

Comment:
Why defining them here and not using True and False? For soft labels? Something else I don't see?

Reply:
No you are right this useful when don't separate the generator loss and the discriminator loss but we do.